src/media_tools/media_import.c

```diff
@@ -7018,6 +7016,237 @@ static GF_Err gf_import_hevc(GF_MediaImporter *import)
 }
 
+void av1_reset_frame_state(AV1StateFrame *frame_state) {
+	if (frame_state->header_obus) {
+		while (gf_list_count(frame_state->header_obus)) {
+			GF_AV1_OBUArrayEntry *a = (GF_AV1_OBUArrayEntry*)gf_list_get(frame_state->header_obus, 0);
+			if (a->obu) gf_free(a->obu);
+			gf_list_rem(frame_state->header_obus, 0);
+			gf_free(a);
+		}
+		gf_list_del(frame_state->header_obus);
+	}
+	
+	if (frame_state->frame_obus) {
+		while (gf_list_count(frame_state->frame_obus)) {
+			GF_AV1_OBUArrayEntry *a = (GF_AV1_OBUArrayEntry*)gf_list_get(frame_state->frame_obus, 0);
+			if (a->obu) gf_free(a->obu);
+			gf_list_rem(frame_state->frame_obus, 0);
+			gf_free(a);
+		}
+		gf_list_del(frame_state->frame_obus);
+	}
+
+	memset(frame_state, 0, sizeof(AV1StateFrame));
+}
+
+static GF_Err gf_import_aom_av1(GF_MediaImporter *import)
+{
+#ifdef GPAC_DISABLE_AV1
+	return GF_NOT_SUPPORTED;
+#endif
+	GF_Err e = GF_OK;
+	GF_AV1Config *av1_cfg = NULL;
+	AV1State state;
+	FILE *mdia = NULL;
+	GF_BitStream *bs = NULL;
+	u32 timescale = 0, dts_inc = 0, track_num = 0, track_id = 0, di = 0, cur_samp = 0;
+	Bool detect_fps;
+	Double FPS = 0.0;
+	u64 pos = 0;
+	typedef enum {
+		OBUs,   /*Section 5*/
+		AnnexB,
+		IVF
+	} AV1BitstreamSyntax;
+	AV1BitstreamSyntax av1_bs_syntax;
+
+	if (import->flags & GF_IMPORT_PROBE_ONLY) {
+		import->nb_tracks = 1;
+		import->tk_info[0].track_num = 1;
+		import->tk_info[0].type = GF_ISOM_MEDIA_VISUAL;
+		import->tk_info[0].flags = GF_IMPORT_OVERRIDE_FPS | GF_IMPORT_FORCE_PACKED;
+		return GF_OK;
+	}
+
+	memset(&state, 0, sizeof(AV1State));
+	av1_cfg = gf_odf_av1_cfg_new();
+	state.config = av1_cfg;
+
+	mdia = gf_fopen(import->in_name, "rb");
+	if (!mdia) return gf_import_message(import, GF_URL_ERROR, "Cannot find file %s", import->in_name);
+
+	detect_fps = GF_TRUE;
+	FPS = (Double)import->video_fps;
+	if (!FPS) {
+		FPS = GF_IMPORT_DEFAULT_FPS;
+	} else {
+		if (import->video_fps == GF_IMPORT_AUTO_FPS) {
+			import->video_fps = GF_IMPORT_DEFAULT_FPS;	/*fps=auto is handled as auto-detection is h264*/
+		} else {
+			/*fps is forced by the caller*/
+			detect_fps = GF_FALSE;
+		}
+	}
+	get_video_timing(FPS, &timescale, &dts_inc);
+
+	bs = gf_bs_from_file(mdia, GF_BITSTREAM_READ);
+
+	track_id = 0;
+	if (import->esd) track_id = import->esd->ESID;
+	track_num = gf_isom_new_track(import->dest, track_id, GF_ISOM_MEDIA_VISUAL, timescale);
+	if (!track_num) {
+		e = gf_isom_last_error(import->dest);
+		goto exit;
+	}
+	gf_isom_set_track_enabled(import->dest, track_num, 1);
+	if (import->esd && !import->esd->ESID) import->esd->ESID = gf_isom_get_track_id(import->dest, track_num);
+	import->final_trackID = gf_isom_get_track_id(import->dest, track_num);
+	if (import->esd && import->esd->dependsOnESID) {
+		gf_isom_set_track_reference(import->dest, track_num, GF_ISOM_REF_DECODE, import->esd->dependsOnESID);
+	}
+	gf_isom_set_cts_packing(import->dest, track_num, GF_TRUE);
+
+	e = gf_media_aom_parse_ivf_file_header(bs, &state);
```

Comment:
Sniffing the type is probably fine for IVF because of the signature but I'm not sure about the AnnexB. Why don't you use the "-add file.av1:fmt=AnnexB" option? with the default being OBU.

Reply:
I've added this in the TODO (see your first msg on this PR). This sniffing worked well on my fuzzing test.